src/nvim/os/env.c

```diff
@@ -76,6 +76,31 @@ char *os_getenvname_at_index(size_t index)
   return name;
 }
 
+/// Check if the environment variable `name` exists and has
+/// a value of either 1, "on", "yes", or "true".
+bool os_env_istruthy(const char *name) FUNC_ATTR_NONNULL_ALL
+{
+  const char *val = os_getenv(name);
+  if (val == NULL || *val == NUL) {  // TODO(Pyrohh): Remove the check for
+    return false;                    // NUL after #2574 is merged.
+  }
+
+  errno = 0;
+  long n = strtol(val, NULL, 10);
+  if (errno == ERANGE || errno == EINVAL) {
```

Comment:
[http://man7.org/linux/man-pages/man3/strtol.3.html]

> The implementation may also set errno to EINVAL in case no conversion was performed (no digits seen, and 0 returned).

We shouldn't consider `EINVAL` an error for cases like `val = "yes"`.